packages/shared/src/components/search/SearchBarInput.tsx

```diff
@@ -108,40 +145,37 @@ function SearchBarInputComponent(
         data-testid="searchBar"
         ref={ref}
       >
-        {sidebarRendered && (
-          <AiIcon
-            size={IconSize.Large}
-            className="mr-3 text-theme-label-tertiary"
-          />
-        )}
-        {!sidebarRendered && (
-          <div className="flex-1 text-theme-label-tertiary">{placeholder}</div>
-        )}
-        {sidebarRendered && (
-          <FieldInput
-            {...inputProps}
-            placeholder={placeholder}
-            ref={inputRef}
-            onFocus={(event) => {
-              onFocus();
-              externalOnFocus?.(event);
-            }}
-            onBlur={(event) => {
-              onBlur();
-              externalOnBlur?.(event);
-            }}
-            onInput={onInput}
-            type="primary"
-            autoComplete="off"
-            className={classNames(
-              'flex-1 caret-theme-status-cabbage',
-              getFieldFontColor({ readOnly, disabled, hasInput, focused }),
-            )}
-          />
-        )}
+        <AiIcon
+          size={IconSize.Large}
+          className="mr-3 text-theme-label-tertiary"
+        />
+        <FieldInput
+          {...inputProps}
+          placeholder={placeholder}
+          ref={inputRef}
+          onFocus={(event) => {
+            onFocus();
+            externalOnFocus?.(event);
+          }}
+          onBlur={(event) => {
+            onBlur();
+            externalOnBlur?.(event);
+          }}
+          onClick={(event) => {
+            onInputClick();
+            externalOnClick?.(event);
+          }}
```

Comment:
Are these actually needed?
I think it's something or field handles no?

Reply:
I think this component was based on the previous Search component. With that, this is to support external and internal handlers.